### PR TITLE
Integrate RN Nightly Build 9/23

### PR DESCRIPTION
**Pull request by chiaramooney**
Commits: https://github.com/facebook/react-native/compare/fa768df2a...d3c7e2018

Changes to Note:
- accessibilityHint now passed through Button: https://github.com/facebook/react-native/commit/be5372801a3ac3ae0f1df76c33f8b5f96ea68828
- Can now specify metro server programmatically: https://github.com/facebook/react-native/commit/e2b5b6504cb4dee8f2ec67f1d3a410801e1ca09a

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/8707)

### CODE REVIEW

**Review by dannyvv — APPROVED** [review]

**Review by jonthysell — APPROVED** [review]

File: packages/playground/windows/playground/packages.config [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="Microsoft.Windows.CppWinRT" version="2.0.210312.4" targetFramework="native"/>
-  <package id="Microsoft.UI.Xaml" version="2.6.0" targetFramework="native"/>
+  <package id="Microsoft.UI.Xaml" version="2.6.1-prerelease.210709001" targetFramework="native"/>
```
[the remark this answers is not in the document]
Comment (chiaramooney) [reply packages/playground/windows/playground/packages.config:4]: 👍 Will revert. Not sure what part of the script caused that update...interesting.

[the remark this answers is not in the document]
Comment (NickGerleman) [reply packages/playground/windows/playground/packages.config:4]: I don't think the integration script touches anything that would modify this. Any chance it was a stale change locally?

File: packages/playground/Samples/text.tsx [hunk 2]
```diff
@@ -11,7 +11,7 @@ export default class Bootstrap extends React.Component {
   render() {
     return (
       <View style={styles.container}>
-        <Text style={styles.welcome}>Welcome to React Native!</Text>
+        <Text style={styles.welcome}>React Native!</Text>
```
Comment (NickGerleman) [inline packages/playground/Samples/text.tsx:14]: What is this change from?

File: packages/playground/windows/playground/packages.config (hunk 1 above)
[the remark this answers is not in the document]
Comment (chiaramooney) [reply packages/playground/windows/playground/packages.config]: Hmm not sure. I re-ran yarn and yarn build and then build playground from the command line.

**Pull request merged**